tests/conftest.py

```diff
@@ -412,21 +411,34 @@
     ),
 ]
 
-# ports for testing, call next() on to avoid assignment overlap
+TEMP_LIBP2P_TEST_DIR = tempfile.mkdtemp()
 DEFAULT_HOST = LOCAL_HOST.hostname
 default_ports = itertools.count(10234)
 
 
-def remove_test_directory(directory: str) -> None:
-    """Destroy a directory once tests are done, change permissions if needed"""
+def remove_test_directory(directory: str, retries: int = 3) -> bool:
+    """Destroy a directory once tests are done, change permissions if needed.
+
+    Note that on Windows directories and files that are open cannot be deleted.
+
+    :param directory: directory to be deleted
+    :param retries: number of re-attempts
+    :return: whether the directory was successfully deleted
+    """
 
     def readonly_handler(func, path, execinfo) -> None:
         """If permission is readonly, we change these and retry."""
         os.chmod(path, stat.S_IWRITE)
         func(path)
 
     # we need `onerror` to deal with permissions, e.g. on Windows
-    shutil.rmtree(directory, onerror=readonly_handler)
+    while os.path.exists(directory) and retries:
+        try:
+            shutil.rmtree(directory, onerror=readonly_handler)
+        except Exception:  # pylint: disable=broad-except
+            retries -= 1
+            time.sleep(1)
+    return not os.path.exists(directory)
```

Comment:
clean-up on Windows is a bit of a pain. Now simply log an error, instead of raising it, if failing to remove the directory after re-attempts. Seemed preferable over `ignore_errors=True` (which would also overwrite the `onerror`)

https://bugs.python.org/issue25024
https://bugs.python.org/issue29982

Any suggestions here are welcome